### PR TITLE
fix: return no session when session id is null or empty string

### PR DESCRIPTION
**Pull request by romansmirnov**
## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: authentication/src/main/java/io/camunda/authentication/session/WebSessionRepository.java [hunk 1]
```diff
@@ -58,15 +58,19 @@ public void save(final WebSession webSession) {
   @Override
   public WebSession findById(final String id) {
     LOGGER.debug("Retrieve session {}", id);
-    return Optional.ofNullable(persistentWebSessionClient.getPersistentWebSession(id))
+    return Optional.ofNullable(id)
+        .map(i -> isSessionIdEmpty(i) ? i : null)
+        .map(persistentWebSessionClient::getPersistentWebSession)
         .map(this::getWebSessionIfNotExpired)
         .orElse(null);
```
Comment (houssain-barouni) [inline authentication/src/main/java/io/camunda/authentication/session/WebSessionRepository.java:65]: I think we can make it simpler this way
```suggestion
    if (isSessionIdEmpty(id)) {
        return null;
    }
    return Optional.ofNullable(persistentWebSessionClient.getPersistentWebSession(id))
        .map(this::getWebSessionIfNotExpired)
        .orElse(null);
```

File: authentication/src/main/java/io/camunda/authentication/session/WebSessionRepository.java [hunk 2]
```diff
@@ -58,15 +58,19 @@ public void save(final WebSession webSession) {
...
   }
 
   @Override
   public void deleteById(final String id) {
     LOGGER.debug("Delete session {}", id);
-    persistentWebSessionClient.deletePersistentWebSession(id);
+    Optional.ofNullable(id)
+        .map(i -> isSessionIdEmpty(i) ? i : null)
```
Comment (houssain-barouni) [inline authentication/src/main/java/io/camunda/authentication/session/WebSessionRepository.java:72]:
```suggestion
        .filter(this::isSessionIdEmpty)
```

**Review by houssain-barouni — COMMENTED** [review]
I added a couple of suggestions to improve readability

**Comment by romansmirnov** [comment]
@houssain-barouni, thanks for the quick review. I now use the `#filter()` option in both cases, which IMO is good enough.

**Review by houssain-barouni — APPROVED** [review]

**Pull request merged**